File: php/tests/JWTBearerTest.php
<?php

namespace Tests\Auth;

use Auth\JWTBearer;
use Firebase\JWT\JWT;
use Firebase\JWT\Key;

class JWTBearerTest extends TestUtils
{
    private $auth;
    private $secretKey = 'test-secret-key';

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new JWTBearer($this->secretKey);
    }

    private function createTestJWT(array $claims = []): string
    {
        $defaultClaims = [
            'sub' => 'test-user',
            'exp' => time() + 3600,
            'iat' => time()
        ];
        $claims = array_merge($defaultClaims, $claims);
        return JWT::encode($claims, $this->secretKey, 'HS256');
    }

    public function testValidJWT(): void
    {
        $token = $this->createTestJWT();
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer ' . $token
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testExpiredJWT(): void
    {
        $token = $this->createTestJWT(['exp' => time() - 3600]);
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer ' . $token
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidSignature(): void
    {
        $token = JWT::encode(['sub' => 'test-user'], 'wrong-secret-key', 'HS256');
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer ' . $token
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMissingToken(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMalformedHeader(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidScheme(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ='
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'jwt');
    }
} 

File: java/src/test/java/com/auth/JWTBearerTest.java
package com.auth;

import com.auth.base.Authentication;
import com.auth.jwtbearer.JWTBearer;
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

class JWTBearerTest extends TestUtils {
    private Authentication auth;
    private String secretKey = "test-secret-key";

    @BeforeEach
    void setUp() {
        super.setUp();
        auth = new JWTBearer(secretKey);
    }

    private String createTestJWT(Map<String, Object> claims) {
        Map<String, Object> defaultClaims = new HashMap<>();
        defaultClaims.put("sub", "test-user");
        defaultClaims.put("exp", new Date(System.currentTimeMillis() + 3600000));
        defaultClaims.put("iat", new Date());
        defaultClaims.putAll(claims);

        return Jwts.builder()
                .setClaims(defaultClaims)
                .signWith(SignatureAlgorithm.HS256, secretKey)
                .compact();
    }

    @Test
    void testValidJWT() {
        String token = createTestJWT(new HashMap<>());
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer " + token);
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testExpiredJWT() {
        Map<String, Object> claims = new HashMap<>();
        claims.put("exp", new Date(System.currentTimeMillis() - 3600000));
        String token = createTestJWT(claims);
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer " + token);
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidSignature() {
        String token = Jwts.builder()
                .setSubject("test-user")
                .signWith(SignatureAlgorithm.HS256, "wrong-secret-key")
                .compact();
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer " + token);
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMissingToken() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMalformedHeader() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidScheme() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ=");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testGetType() {
        assertAuthType(auth, "jwt");
    }
} 